pkg/operator/controllers/machinehealthcheck/machinehealthcheck_controller.go

```diff
@@ -0,0 +1,98 @@
+package machinehealthcheck
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"context"
+
+	machinev1beta1 "github.com/openshift/machine-api-operator/pkg/apis/machine/v1beta1"
+	"github.com/sirupsen/logrus"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	kruntime "k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/kubernetes/scheme"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+
+	arov1alpha1 "github.com/Azure/ARO-RP/pkg/operator/apis/aro.openshift.io/v1alpha1"
+	aroclient "github.com/Azure/ARO-RP/pkg/operator/clientset/versioned"
+	"github.com/Azure/ARO-RP/pkg/operator/controllers"
+	"github.com/Azure/ARO-RP/pkg/util/dynamichelper"
+)
+
+const (
+	CONFIG_NAMESPACE string = "aro.machinehealthcheck"
+	ENABLED          string = CONFIG_NAMESPACE + ".enabled"
+)
+
+type Reconciler struct {
+	log *logrus.Entry
+
+	arocli aroclient.Interface
+	dh     dynamichelper.Interface
+}
+
+func NewReconciler(log *logrus.Entry, arocli aroclient.Interface, dh dynamichelper.Interface) *Reconciler {
+	return &Reconciler{
+		log:    log,
+		arocli: arocli,
+		dh:     dh,
+	}
+}
+
+func (r *Reconciler) Reconcile(ctx context.Context, request ctrl.Request) (ctrl.Result, error) {
+	instance, err := r.arocli.AroV1alpha1().Clusters().Get(ctx, arov1alpha1.SingletonClusterName, metav1.GetOptions{})
+	if err != nil {
+		return reconcile.Result{}, err
+	}
+
+	if !instance.Spec.OperatorFlags.GetSimpleBoolean(ENABLED) {
+		// controller is disabled
+		return reconcile.Result{}, nil
+	}
+
+	var resources []kruntime.Object
+	for _, assetName := range AssetNames() {
+		b, err := Asset(assetName)
+		if err != nil {
+			r.log.Error(err)
```

Comment:
No need for the error logging here, since you're returning it on the next line. The reconciler should return any errors if they occur.